theano/gof/tests/test_utils.py

```diff
@@ -61,3 +61,44 @@ def test_stack_trace():
         assert len(v.tag.trace[0]) == 2
     finally:
         theano.config.traceback.limit = orig
+
+
+class TestCaching():
+    class CachingClassExample(theano.Op):
+        __props__ = ('value',)
+
+        def __init__(self, value):
+            self.value = value
+
+    class CachingClassExample_2(theano.Op):
+        __props__ = ('value',)
+
+        def __init__(self, var):
+            self.var = var
+
+    class CachingClassExample_3(theano.Op):
+        __props__ = ()
+
+        def __init__(self, var):
+            self.var = var
+
+    class CachingClassExample_4(theano.Op):
+        def __init__(self, var):
+            self.var = var
+
+    def test_caching(self):
+        obj_1 = self.CachingClassExample(3)
+        obj_2 = self.CachingClassExample(3)
+        assert obj_1 is obj_2
+
+        obj_1 = self.CachingClassExample_2(3)
+        obj_2 = self.CachingClassExample_2(3)
+        assert obj_1 is not obj_2
+
+        obj_1 = self.CachingClassExample_3(3)
+        obj_2 = self.CachingClassExample_3(3)
+        assert obj_1 is obj_2
```

Comment:
I think this should not be cached. I think this is the cause that the memory error still continue.